posts/2021-08-01-disjoint-capture-in-closures.md

```diff
@@ -0,0 +1,62 @@
+
+Like any Edition migration, we have also created lints that will warn you of upcoming changes and suggest precise edits to preserve the semantics of your code. This blog post will explain the new feature and also tell you how you can try it out today on nightly if you like.
+
+
+## What are closures?
+Closures are anonymous functions you can save in a variable or pass as an argument to other functions. They are referred to as ”lambda functions” in other programming languages.
+    
+This is a basic closure that prints "Hello, World!":
+
+```rust
+let c = || println!("Hello, World!"); 
+```
+
+
+Closures can capture variables from the scope they are defined in.
+ ```rust
+let my_string = "Hello, World";
+let c = || println!("{}", my_string); // closure captures (& my_string)
+```
+
+
+If you're new to Rust and haven't used closures, we recommend you check out the [Rust Book](https://doc.rust-lang.org/book/ch13-01-closures.html) and [Rust by Example](https://doc.rust-lang.org/rust-by-example/fn/closures.html) to learn how to use closures.
+
+For those who would like a quick recap, we recommend reading the [Rust Reference](https://doc.rust-lang.org/reference/types/closure.html).
+
+
+## What is disjoint capture?
+
+Disjoint capture is the ability of a closure to capture only part of a variable rather than the variable in its entirety (Rust 2018 behavior). The goal with this feature is to unify the semantics of closure captures with those of borrowing and move operations.
+
+Consider a Point `p` with fields `x` and `y` corresponding to the x and y coordinates of the point.
+
+```rust
+let p = Point { x: 10, y: 20 };
+
+let c = || println!("{}", p.x);
+```
+
+`c` in Rust 2018 will capture `p` completely, but in Rust 2021 it will only capture `p.x`. This now matches what gets borrowed when someone writes `println!("{}", p.x)`.
+
+Disjoint capture was proposed as part of [RFC 2229](https://github.com/rust-lang/rfcs/blob/master/text/2229-capture-disjoint-fields.md), and we suggest reading the RFC to better understand motivations behind the feature.
+
+The precise path that gets captured is typically the full path that is used in the closure, but there are cases where we will only capture a prefix of the path. See this pending PR to the [Rust Reference](https://github.com/rust-lang/reference/blob/d24c4642c8efd26915209558e065a5e670363fc0/src/types/closure.md) for the full details.
+
+The feature also includes (minor) breaking changes to the Rust semantics which are also documented in this pending PR to the [Rust Reference](https://github.com/rust-lang/reference/blob/d24c4642c8efd26915209558e065a5e670363fc0/src/types/closure.md).
```

Comment:
Maybe link to the PR https://github.com/rust-lang/reference/pull/1059 instead of a commit, since this commit was force pushed over AFAICT.